core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -795,7 +796,8 @@ private[spark] object JsonProtocol {
       case `taskKilled` => TaskKilled
       case `executorLostFailure` =>
         val executorId = Utils.jsonOption(json \ "Executor ID").map(_.extract[String])
-        ExecutorLostFailure(executorId.getOrElse("Unknown"))
+        val reason = Utils.jsonOption(json \ "Loss Reason").map(_.extract[String])
+        ExecutorLostFailure(executorId.getOrElse("Unknown"), reason.getOrElse("Unknown"))
```

Comment:
Hmm... you'll also need to change `taskEndReasonFromJson`, otherwise the history server won't see the new information. Also, when adding new properties to serialized classes, we generally use `Option[Foo]` (see calls to `Utils.jsonOption` in this class).